voluncheer/profiles/views/profile.py

```diff
@@ -200,15 +200,11 @@ def confirm_attendance(request, opportunity_id):
         volunteer = Volunteer.objects.get(pk=attendee_pk)
         if volunteer not in opportunity.volunteers.all():
             continue
-        if volunteer in opportunity.attended_volunteers.all():
-            opportunity.attended_volunteers.remove(volunteer)
-            volunteer.hours_volunteered -= opportunity.duration
-            volunteer.save()
-        else:
-            opportunity.attended_volunteers.add(volunteer)
-            volunteer.hours_volunteered += opportunity.duration
-            volunteer.save()
-
+        opportunity.attended_volunteers.add(volunteer)
```

Comment:
That's fair but if that's the case we have a usability issue. This would require the person confirming attendance to confirm/deny the attendance of _every_ volunteer _every_ time they perform this action. Scale that out to O(100) people and no one is going to use it.

That said, I don't have a great way to un-record the attendance of a participant but I also don't think it belongs here.

Thoughts?

Reply:
Yeah I agree. I don't think the organization would diligent unrecord volunteers at least by our current design.

Reply:
> > > I tried out the code locally, however, I still see an additional 20 hours added to the calculations. My volunteer started with 3 hours, and it should have added an opportunity for only 2 hours and 45 min. Instead, it added 22 hours and 45 min. I think the bug might have to do with the opportunity.duration calculation instead?
> > 
> > 
> > OK, I figured out the problem but don't yet have a solution. The `TimeField` is inherently non-timezone aware and there appears to be no option for this. So I need to think more on this if we want to keep the field `time` exclusively and not a datetime.
> 
> I think Doug's PR just makes duration of opportunity timezone aware. This might help you to more accurately account for hours.

I think Doug essentially fixed this. The issue was timezone awareness and he made it timezone aware.

Reply:
An organization does not have to confirm/deny the volunteer's attendance every time they use it. They only need to deny a volunteer's attendance if they had already confirmed it first and their sticky note is green.